test/constrained-generics/ucol/cwailes-1/tier_1_subtyping_0.chpl

```diff
@@ -16,5 +16,6 @@ proc f(x:?T):int where implements I(T) {
   return g(x);
 }
 
+// Infers 'implements I(borrowed D);'
 // Prints 42
-writeln(f(new D(42)));
+writeln(f(new borrowed D(42)));
```

Comment:
I would think with your conversions we can have f accept owned D now. Anyway I think we are trying to avoid `new borrowed`.

If new owned doesn't work for some reason that's one to make a future about I think. I see the note about the future below but this looks like a different case to me (it is not returning the owned)

Reply:
Indeed, the `owned D` variant does not work, nor does `shared D`, due to #8489. I left this test with a `borrowed` to illustrate something that works with subtyping. I added a separate .future for `owned` and `shared`.